### PR TITLE
[breaking] Upgrade libraries to TF 0.12

### PR DESCRIPTION
**Pull request by mbarrien**
### Summary
This PR makes Fogg support Terraform 0.12 syntax (and only 0.12 syntax). The different HCL2 syntax makes for a required breaking change to properly support it.

### Test Plan
Passes existing unit tests.

### CODE REVIEW

File: testdata/bless_provider/terraform/accounts/foo/fogg.tf [hunk 1]
```diff
@@ -1,12 +1,20 @@
 # Auto-generated by fogg. Do not edit
 # Make improvements in fogg, so that everyone can benefit.
 
+
```
Comment (ryanking) [inline testdata/bless_provider/terraform/accounts/foo/fogg.tf]: why do we get all these blank lines now?

File: testdata/v1_full/terraform/envs/stage/cloud-env/main.tf [hunk 2]
```diff
@@ -5,4 +5,5 @@ module "fogg-test-module" {
   source = "github.com/chanzuckerberg/fogg-test-module"
   bar    = "${local.bar}"
   foo    = "${local.foo}"
+
```
Comment (ryanking) [inline testdata/v1_full/terraform/envs/stage/cloud-env/main.tf]: is this a change in tf fmt?

File: testdata/bless_provider/terraform/accounts/foo/fogg.tf (hunk 1 above)
Comment (mbarrien) [reply testdata/bless_provider/terraform/accounts/foo/fogg.tf]: Terraform fmt seems to add way more blank lines. Don't know why, even though the open issue says it won't add new lines.

https://github.com/hashicorp/terraform/issues/21360#issuecomment-493807349

File: testdata/v1_full/terraform/envs/stage/cloud-env/main.tf (hunk 2 above)
Comment (mbarrien) [reply testdata/v1_full/terraform/envs/stage/cloud-env/main.tf]: Same as above.

File: testdata/bless_provider/terraform/accounts/foo/fogg.tf (hunk 1 above)
Comment (ryanking) [reply testdata/bless_provider/terraform/accounts/foo/fogg.tf]: I think that comment is mislead, it seems that they only care about horizontal spacing.

**Pull request merged**